### ISSUE TITLE
Dynamic zone components for specific local can't call up via API GET Request

### ISSUE THREAD

**Issue by CihanNewSenses** [strapi/strapi#22447]
### Required System information

- Node.js version: v21.1.0
- NPM version: 10.2.0
- Strapi version: v5.5.1
- Database: sqlite
- OS: darwin-x64
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Local values from dynamic zones components can't called up via API. It only shows the values from default local, but not from specific local. I used standard get request /api/:pluralApiId?populate=* in localhost.

### Steps to reproduce the behavior

1. Go to 'Postmen App or Postmen Browser App'
2. Create Get Request with  /api/:pluralApiId?populate=*
3. See error

### Expected behavior

Local dynamic zones components values should be displayed in the localizations data object like any other fields.

### Screenshots

![strapi_screenshot2](https://github.com/user-attachments/assets/f7491122-872d-47b8-8031-2dc285e6f275)
![strapi_screenshot1](https://github.com/user-attachments/assets/eb3c9ea7-2fce-4738-91ea-ab1f5053a81a)

### Response Example

`{
    "data": [
        {
            "id": 11,
            "documentId": "bzs3m8ejgjum484ew05zdqey",
            "testField": "TestValue1",
            "testField2": "TestValue2",
            "createdAt": "2024-12-16T12:19:15.240Z",
            "updatedAt": "2024-12-16T12:46:54.708Z",
            "publishedAt": "2024-12-16T12:46:54.718Z",
            "locale": "en",
            "local": [
                {
                    "__component": "test.test",
                    "id": 6,
                    "text_field": "dynamictext Default local"
                }
            ],
            "localizations": [
                {
                    "id": 10,
                    "documentId": "bzs3m8ejgjum484ew05zdqey",
                    "testField": "TestValue1",
                    "testField2": "TestValue2",
                    "createdAt": "2024-12-16T12:46:31.158Z",
                    "updatedAt": "2024-12-16T12:46:54.725Z",
                    "publishedAt": "2024-12-16T12:46:48.578Z",
                    "locale": "de"
                }
            ]
        }
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "pageSize": 25,
            "pageCount": 1,
            "total": 1
        }
    }
}
`

**Comment by derrickmehaffy** [comment]
TLDR: The localization array isn't using the same query params for population. I'm honestly not sure if that's intended as I would expect the best way to do this would be to fetch the specific localized entity directly or add the `locale` query param to fetch the other language.

I'll leave this here to get @alexandrebodin 's input since he built this.
